MAIN-SBP-CSP-UpdateInfra: Update for SLES 15 SP3

mfriesenegger opened the pull request.

lproven commented:
Your indentation is very strange and makes this a little hard to read. I suggest running `daps-xmlformat -i $FILENAME` over it to re-indent it. Our house style is: 
* indent by 1 space only;
* opening and closing tags on the same indentation level;
* [screen] tags not indented at all, whatever the indent of the closing text.

lproven commented on xml/MAIN-SBP-CSP-UpdateInfra.xml:
```diff
@@ -389,7 +398,7 @@ ssh-keygen -t rsa -f regionsrv</screen>
 					</step>
 					<step>
-						<para> Enter your organizations credentials from SCC. </para>
+						<para> Enter your organizations credentials from SCC and disable forwarding of registrations. </para>
```
```suggestion
						<para> Enter your organization's credentials from SCC and disable forwarding of registrations. </para>
```

lproven commented on xml/MAIN-SBP-CSP-UpdateInfra.xml:
```diff
@@ -413,6 +422,107 @@ ssh-keygen -t rsa -f regionsrv</screen>
+				<para> After the installation of the RMT server package, the directory structure
+					required by RMT is set up in <filename>/var/lib/rmt/public/repo</filename>. The
+					repositories should <emphasis role="strong">not</emphasis> be stored on the root
+					volume. Storing the repository cash on the root volume will significantly
```
```suggestion
					volume. Storing the repository cache on the root volume will significantly
```

lproven commented on xml/MAIN-SBP-CSP-UpdateInfra.xml:
```diff
@@ -413,6 +422,107 @@ ssh-keygen -t rsa -f regionsrv</screen>
+					increase the recovery time should the RMT system experience issues and need to
+					be re-created. The following process outlines the steps necessary to set up the
+					external device to hold the repositories. </para>
+				<para> For the RMT repository disk follow the steps outlined below: </para>
```
```suggestion
				<para> For the RMT repository disk, follow the steps outlined below: </para>
```

lproven commented on xml/MAIN-SBP-CSP-UpdateInfra.xml:
```diff
@@ -413,6 +422,107 @@ ssh-keygen -t rsa -f regionsrv</screen>
+							the RMT server. </para>
+					</step>
+					<step>
+						<para> Use <command>sudo fdisk -l</command> to list the available block
+							devices. </para>
```
It seems to me that this assumes MBR disks, i.e. volumes under 2TB. This is **not** a safe assumption any longer in 2021.

mfriesenegger replied:
Changed to "sudo lsblk".

rjschwei commented:
Also what I am missing is a change to the over all introduction that was discussed with @mfriesenegger where we agreed that we wanted to make it more explicit about how the puzzle pieces fit together. Something like this:

"""
The described setup presents a system of interacting components. These components are the Region Server, the update server, and the region server client which works in the instance the user launches. The system as implemented and provided by SUSE requires that all components of the system be implemented as described. The system is cloud framework implementation agnostic and as such can be implemented for any framework implementation. This does not imply that the system must be set up in the described way. However if components are modified or alternative architectures are used the parts provided by SUSE will no longer function in a "plug and play" fashion and it is up to the CSP to implement the necessary components to fit the architectural changes being made.
"""

mfriesenegger commented on xml/MAIN-SBP-CSP-UpdateInfra.xml:
```diff
@@ -40,7 +39,7 @@
+		<date>September 03, 2021</date>
 		<abstract>
 			<para> This guide describes the setup of the recommended infrastructure for offering
 				SUSE product as on-demand offerings in a cloud environment. The update
```
I will do what @rjschwei suggests.  I was able to find &sles;.  Is there one for SLES for SAP Applications?

mfriesenegger replied:
&sles; does not work when I ran daps.  For now I spelled out SUSE Linux Enterprise Server and SUSE Linux Enterprise Server for SAP Applications.

lproven replied:
Entity names can be found in `xml/entity-decl.ent`.

SLES:
\<!ENTITY sls           "SUSE Linux Enterprise Server"\>
\<!ENTITY slsreg        "SUSE\&reg; Linux Enterprise Server"\>

Generally, the first time one uses the name, one should use the version with the ® symbol; after that, the version without.

I can't see one for SLES for SAP, but we can easily add one.

mfriesenegger replied:
This is now using entity names &sls; and &slssap;.  I added &slssap; to xml/entity-decl.ent.  I am having difficulty using &slsreg; and &slssapreg; which generate an error during document creation.  @lproven, can you help fix this?

mfriesenegger commented on xml/MAIN-SBP-CSP-UpdateInfra.xml:
```diff
@@ -119,11 +118,18 @@
+					registration server. </para>
+				<para>
+					If a cloud provider chooses to not use the automatic RMT selection mechanism in the
+					region server then the cloud provider must replace the a susecloud url resolver for
```
Will you accept that the sentence after "susecloud url resolver plugin for zypper" is introduced explains what "susecloud url resolver plugin for zypper" means?

lproven replied:
It's not just that. :-)

So... product names should use entities wherever possible, for 2 reasons.

1. We rename products sometimes. :-) For instance, CaaSP went through several in my time with SUSE.
2. It means the name will always look the same and match other usages. This eliminates the chances of typos etc.

If **I'm** not sure what you mean then readers may not be.

Are you referring to:

    <!ENTITY cloudreg       "&suse;&reg; Cloud">
    <!ENTITY cloud          "&suse; OpenStack Cloud">

...?

If you're talking about Zypper the packaging tool, then it's a name, so capitalize it. URL is an acronym or initialism depending on who you ask ("you are ell" or "earl") so use block caps.

So instead of:

> the cloud provider must replace the a susecloud url resolver for zypper

It should probably say something like:

"... The cloud provider should replace **the** _&cloud;_ **URL** resolver for **Zypper**..."

mfriesenegger replied:
I think there is confusion which I will try to clear up.

- Zypper has a plugin capability  called "Url Resolver plugins" - https://doc.opensuse.org/projects/libzypp/HEAD/zypp-plugins.html#plugin-url-resolver 
- There is a Zypper Url Resolver plugin called "susecloud"
- The "susecloud" plugin is not related to any products like SUSE Cloud or SUSE OpenStack Cloud.
- @rjschwei in another comment in this PR asked that we use "Url Resolver" to be consistent with https://doc.opensuse.org/projects/libzypp/HEAD/zypp-plugins.html#plugin-url-resolver

I believe I have done this correctly.

mfriesenegger replied:
I was using <command>zypper</command> but have changed to using Zypper because this is referring to the tool and not the command.

lproven replied:
> I think there is confusion which I will try to clear up.

There was!

>     * Zypper has a plugin capability  called "Url Resolver plugins" - https://doc.opensuse.org/projects/libzypp/HEAD/zypp-plugins.html#plugin-url-resolver

Aha! OK.

>     * There is a Zypper Url Resolver plugin called "susecloud"

OK. What is it or what does it do?

>     * The "susecloud" plugin is not related to any products like SUSE Cloud or SUSE OpenStack Cloud.

Got it. Fine, no problem -- consider my suggestion to use entities withdrawn.

>     * @rjschwei in another comment in this PR asked that we use "Url Resolver" to be consistent with https://doc.opensuse.org/projects/libzypp/HEAD/zypp-plugins.html#plugin-url-resolver

:-( Ouch. This one I do **not** like. Sorry, @rjschwei, but I find that ugly. "Url" implies it's a name or a word; it is 
not, it is an initialism for Universal Resource Locator. "Url" is IMHO wrong and it should always be "URL". "Resolver" is not a proper noun and  does not need to be capitalized.

Consistency is good, but if the openSUSE docs team do not try to be consistent with the SUSE Style Guide, I do not think we need to be consistent with them.

> I believe I have done this correctly.

Fair. I think so too, now that the context is clearer to me.

mfriesenegger replied:
> > ```
> > * There is a Zypper Url Resolver plugin called "susecloud"
> > ```
> 
> 
> OK. What is it or what does it do?

The susecloud zypper plugin is part of the cloud-regionsrv-client package that is installed in a SLES client image that is deployed as an instance within a cloud providers infrastructure.  It provides the URL to the closest RMT servers with a cloud providers infrastructure.

I am treating my portion of the conversation as resolved but will not mark it resolved so @rjschwei can review and comment.

mfriesenegger commented:
> The system as implemented and provided by SUSE requires that all components of the system be implemented as described. The system is cloud framework implementation agnostic and as such can be implemented for any framework implementation. This does not imply that the system must be set up in the described way. However if components are modified or alternative architectures are used the parts provided by SUSE will no longer function in a "plug and play" fashion and it is up to the CSP to implement the necessary components to fit the architectural changes being made.

What I wrote "If a cloud provider chooses not to use the automatic RMT selection mechanism..." was my attempt based on our conversation.  I like what you have written and have added with a few changes to the introduction.

chabowski reviewed (commented):
Hi Mike, I am not requesting any changes here in this version myself. But have the proposed changes from Robert and Liam been integrated? If yes, I would merge the PR into the official SBP repo and do a final edit/review there and publish the paper from there. THANKS!

mfriesenegger commented:
> Hi Mike, I am not requesting any changes here in this version myself. But have the proposed changes from Robert and Liam been integrated? If yes, I would merge the PR into the official SBP repo and do a final edit/review there and publish the paper from there. THANKS!

Hello Meike.  There are several remaining comments which should be reviewed by @lproven and @rjschwei before publishing.  Thank you.

mfriesenegger commented:
> Hello Meike. There are several remaining comments which should be reviewed by @lproven and @rjschwei before publishing. Thank you.

I think all of the @lproven requests have been handled.  I have not heard from @rjschwei so I think this should wait to be published.

rjschwei commented on xml/MAIN-SBP-CSP-UpdateInfra.xml:
```diff
@@ -7,469 +7,765 @@
+      <filename>/usr/lib/systemd/system/rmt-server-sync.timer</filename>. In
+      addition you want to have a generous value for the
+      <emphasis role="strong"
+							>RandomizedDelaySec</emphasis>. SUSE
```
s/SUSE/&suse;/

rjschwei commented on xml/MAIN-SBP-CSP-UpdateInfra.xml:
```diff
@@ -483,417 +779,329 @@ ssh-keygen -t rsa -f regionsrv</screen>
+    <listitem>
+     <para>
+      Two or more RMT servers need to be available in a region that have been
+      deployed using SUSE Linux Enterprise Server 15 SP3.
```
see earlier comment about version specific reference. This implies you have to update the document for SP4, then SP5 and then SP6.....

rjschwei commented on xml/MAIN-SBP-CSP-UpdateInfra.xml:
```diff
@@ -948,267 +1164,323 @@ regsharing:
+    server placement could be based on geographic boundaries like Americas,
+    Europe/Middle East and Asia. It is recommended that a minimum of three
+    region servers are deployed for redundancy. Install, register and fully
+    patch SUSE Linux Enterprise Server 15 SP3 for each of the region servers.
```
SLES version specific

mfriesenegger replied:
There are other places in the document where SUSE Linux Enterprise Server 15 SP3 is used.  I am not going to change the SLES version specific information because this shows the reader that the document is being updated for newer versions.

rjschwei commented on xml/MAIN-SBP-CSP-UpdateInfra.xml:
```diff
@@ -1217,267 +1489,384 @@ systemctl enable rmt-server-regsharing.timer
+      octet (254 IP addresses). The Region Server also creates a secondary hash
+      table that relates the region names to the RMT server information. The
+      secondary hash table is consulted if the Region Server receives a region
+      hint from the client.
```
The implementation of the region server has changed. We should probably leave out the details. Suggest:

```The Region Server reads the <filename>regionData.cfg</filename> file as
configured in the <filename>regionInfo.cfg</filename> file at start-up. When a client requests information the provided region hint from the client data is used as the section name in <filename>regionData.cfg</filename> and teh update server data is returned. If no region hint is provided a longest prefix match is used to attempt a client IP look up to match the update server data to the client.```

rjschwei commented on xml/MAIN-SBP-CSP-UpdateInfra.xml:
```diff
@@ -1217,267 +1489,384 @@ systemctl enable rmt-server-regsharing.timer
+    <listitem>
+     <para>
+      The public key files for all region servers generated with the
+      <command>genRegionServerCert</command> command. File names must be in
```
s/File names/Files/

rjschwei commented on xml/MAIN-SBP-CSP-UpdateInfra.xml:
```diff
@@ -1217,267 +1489,384 @@ systemctl enable rmt-server-regsharing.timer
+    </listitem>
+    <listitem>
+     <para>
+      The configuration file for the client
```
s/client/client is/ ?

rjschwei commented on xml/MAIN-SBP-CSP-UpdateInfra.xml:
```diff
@@ -1217,267 +1489,384 @@ systemctl enable rmt-server-regsharing.timer
+     <para>
+      The configuration file for the client
+      <filename>/etc/regionserverclnt.cfg</filename>. The configuration file is
+      <filename>init</filename> format.
```
Previously the format was given with ```<emphasis>``` tagging

chabowski commented on xml/MAIN-SBP-CSP-UpdateInfra.xml:
```diff
@@ -7,469 +7,765 @@
+<!--    <jobtitle>Solution Architect</jobtitle>-->
+<!--    <orgname>&suse;</orgname>-->
+<!--   </affiliation>-->
+  </author><date>September 03, 2021</date>
```
@rjschwei Hi Robert,
it is a bit of a challenge if authors leave SUSE - but are not responsible anymore - to still list their names on the paper - and this is for several reasons: 

- One of the main reasons why we add (SUSE) author names to best practices is to know whom to approach if changes or updates are needed. If an author left - this information is useless at least for this purpose
- If we would keep authors listed who left SUSE, after an update of the paper, would we list them with their previous SUSE job function, or with their new company and job function? The first option would simply be wrong, the second option is not really practicable because we would need to ask for permission from their new company
- Also,  if an author inside SUSE changes roles, and another one takes over subject matter expertise, and from now on is "responsible" for a tech document, should we have the colleague NOT being responsible anymore still listed? If responsibilities change often, the list of authors could become veeeeeery long.
- Same is valid in general for (shorter) technical docs referring to a specific product version / SP / software, as usually their "half-life period" is also much shorter than for example for books, and updates are happening more frequently...

Thus, we decided to list only the *actual* or *currently* responsible authors and SME's. For SUSE papers, as everything written during work time and with SUSE equipment is SUSE property anyway, we would not need to list authors. But we do that for the SBP's because we think it is just fair to give them at least some credits for the creation of a document, and for their (current) work on a technical solution.

Hope this explains it better. 
Thanks very much - Meike

rjschwei replied:
Yes, I understand all the reasoning, which is why referring to teams is a great why to avoid having to follow the bouncing ball of personal career choices.

lproven reviewed (approved):
I can reformat this after it is accepted, if that is easier, rather than muddling up the PR history by doing it now.

mfriesenegger commented:
> I can reformat this after it is accepted, if that is easier, rather than muddling up the PR history by doing it now.

@chabowski is planning to work on this on Monday.  I am fine with whatever you and she decided to do to prepare this for publishing.  Thank you!

rjschwei commented:
Well unless GitHub is playing tricks on me, which it has done previously, the current diff doesn't look like all concerns I reaise previously have been addressed.

mfriesenegger commented:
> Well unless GitHub is playing tricks on me, which it has done previously, the current diff doesn't look like all concerns I reaise previously have been addressed.

I think I have addressed all of your concerns either by making the change you recommended or replying to your comment why I felt that a a change should not be made.  I could have messed up my pushes with the latest updates.  Can you give an example that I can look at?   Thanks in advance.

rjschwei commented:
> > Well unless GitHub is playing tricks on me, which it has done previously, the current diff doesn't look like all concerns I reaise previously have been addressed.
> 
> I think I have addressed all of your concerns either by making the change you recommended or replying to your comment why I felt that a a change should not be made. I could have messed up my pushes with the latest updates. Can you give an example that I can look at? Thanks in advance.

When I look at the diff via https://github.com/SUSE/suse-best-practices/pull/214/files I still see the explanation on line 157 for which I made a suggestions  to logically better separate the URL resolver form the region server implementation. I do see that you marked the comments as resolved but I do not see the updated file for some reason. Again, could be a GitHub issue.

mfriesenegger commented:
> I do see that you marked the comments as resolved but I do not see the updated file for some reason. Again, could be a GitHub issue.

I thought I had resolved all comments but I see a number of your comments that are not resolved after expanding all of your comments looking for comments dealing with line 157.  I do not know for certain how I skipped these but I will take care of them now.

mfriesenegger commented:
@rjschwei My apologizes for the confusion.  I believe all your comments are incorporated and resolved.  Thank you for your detailed review of the document!  I am requested a final review by everyone.

chabowski reviewed (approved):
I will now merge this pull request into the main repo. All, please think of refreshing your main fork before you do additional changes. Please ensure you have main updated to the latest status. Thanks!

chabowski commented:
> I can reformat this after it is accepted, if that is easier, rather than muddling up the PR history by doing it now.

@lproven Thank you very much for your review. I am taking over from here. Will merge now and perform final review via style checker etc.  Will also run the indentation command (if necessary - Frank said DAPS would do it automatically for xml docs ...). Thanks again!

chabowski commented:
@mfriesenegger here is the latest PDF version for a final check. I did some edits (e.g. we moved to sentence-style section titles a while ago, no capitalization anymore ..., some typo/punctuation fixing, adjusting titles to our style, etc.) but it should be all okay.
Please have a look: 
[SBP-CSP-UpdateInfra_color_en.pdf](https://github.com/SUSE/suse-best-practices/files/7155788/SBP-CSP-UpdateInfra_color_en.pdf)

mfriesenegger commented:
> @mfriesenegger here is the latest PDF version for a final check. I did some edits (e.g. we moved to sentence-style section titles a while ago, no capitalization anymore ..., some typo/punctuation fixing, adjusting titles to our style, etc.) but it should be all okay.
> Please have a look:
> [SBP-CSP-UpdateInfra_color_en.pdf](https://github.com/SUSE/suse-best-practices/files/7155788/SBP-CSP-UpdateInfra_color_en.pdf)

Thank you @chabowski.  I reviewed and found a few minor formatting items to fix which I sent to you in an email.  Let me know if you prefer that I make the changes or if you are willing to do this.

chabowski commented:
> > @mfriesenegger here is the latest PDF version for a final check. I did some edits (e.g. we moved to sentence-style section titles a while ago, no capitalization anymore ..., some typo/punctuation fixing, adjusting titles to our style, etc.) but it should be all okay.
> > Please have a look:
> > [SBP-CSP-UpdateInfra_color_en.pdf](https://github.com/SUSE/suse-best-practices/files/7155788/SBP-CSP-UpdateInfra_color_en.pdf)
> 
> Thank you @chabowski. I reviewed and found a few minor formatting items to fix which I sent to you in an email. Let me know if you prefer that I make the changes or if you are willing to do this.

Hi @mfriesenegger welcome back ;-). I am currently in the Nuremberg office, but will drive home soon. When I am back (in about an our or so), I can add the changes. But if you prefer, you can also do them yourself via a PR. Just let me know.

mfriesenegger commented:
> Hi @mfriesenegger welcome back ;-). I am currently in the Nuremberg office, but will drive home soon. When I am back (in about an our or so), I can add the changes. But if you prefer, you can also do them yourself via a PR. Just let me know.

Thank you!  I decided to make the changes and submit https://github.com/SUSE/suse-best-practices/pull/217 for you to review and merge.
